### ISSUE TITLE
Can't access `toggleterm` module from within the `config.lua` file

### ISSUE THREAD

**Issue by josemiguelo** [LunarVim/LunarVim#3091]
### Problem description

_I'm a complete newbie at LunarVim and Lua in general, so please forgive me if the question is dumb._ 

When I require toggleterm (`require("toggleterm")`) from within the `config.lua` file, I get the following message:

`lvim: "Invalid configuration: /Users/user/.config/lvim/config.lua:1: module 'toggleterm' not found:\n\tno field package.preload['toggleterm']\n\tno file './toggleterm.lua'\n\t
no file '/usr/local/share/luajit-2.1.0-beta3/toggleterm.lua'\n\tno file '/usr/local/share/lua/5.1/toggleterm.lua'\n\tno file '/usr/local/share/lua/5.1/toggleterm/init.lua'\n\tno file '/usr/local/sha
re/lua/5.1/toggleterm.lua'\n\tno file '/usr/local/share/lua/5.1/toggleterm/init.lua'\n\tno file './toggleterm.so'\n\tno file '/usr/local/lib/lua/5.1/toggleterm.so'\n\tno file '/usr/local/lib/lua/5.1
/toggleterm.so'\n\tno file '/usr/local/lib/lua/5.1/loadall.so'" file="init.lua", line=119`

but I do can access some other plugins from within this file, for example `require("telescope")` 

Also I notice with autocompletion I can access al modules defined in Lvim's core except for toggleterm. What might be the reason? I know Lunarvim uses the [plugin](https://github.com/LunarVim/LunarVim/blob/master/lua/lvim/plugins.lua#L249)

### LunarVim version

master-e4a5fe9

### Neovim version (>= 0.7.2)

0.7.2

### Operating system/version

MacOS 12.5.1

### Steps to reproduce

Steps to reproduce:
1.  Install the stable branch of LunarVim
2. Add `require("toggleterm")` to the `config.lua` file
3. Execute `PackerCompile`,  `LvimCacheReset` and restart.
4. Watch the error

### support info

```console
 Language client log: /Users/user/.cache/lvim/lsp.log
 Detected filetype:   lua
 
 1 client(s) attached to this buffer: 
 
 Client: sumneko_lua (id: 1, pid: 54511, bufnr: [1])
 	filetypes:       lua
 	autostart:       true
 	root directory:  Running in single file mode.
 	cmd:             lua-language-server
 
 Configured servers list: sumneko_lua
```

```console
Buffer info
* filetype:                lua
* bufnr:                   1
* treesitter status:       active

Active client(s)
* name:                      sumneko_lua
* id:                        1
* supported filetype(s):     lua
* attached buffers:          1
* root_dir pattern:          1
* capabilities:              foldingRangeProvider    | colorProvider          | documentHighlightProvider
                             workspaceSymbolProvider | documentSymbolProvider | documentFormattingProvider
                             typeDefinitionProvider  | referencesProvider     | documentRangeFormattingProvider
                             definitionProvider      | hoverProvider

Automatic LSP info
* Skipped servers: [angularls, ansiblels, csharp_ls, cssmodules_ls, denols, ember, emmet_ls, eslint, golangci_lint_ls, graphql, jedi_language_server, ltex, phpactor, psalm, pylsp, quick_lint_js, rome, reason_ls, solang, sorbet, sourcery, sqlls, sqls, stylelint_lsp, tflint, svlangserver, verible, vuels]

Formatters info
* Active: 
* Supported: [lua_format, stylua]

Linters info
* Active: 
* Supported: [luacheck, selene]

Code actions info
* Active: 

```

**Comment by lvimuser** [comment]
For [core plugins](https://www.lunarvim.org/plugins/#core-plugins) prefer the `lvim.builtin.<plugin>` interface.

> When I require toggleterm (`require("toggleterm")`) from within the `config.lua` file, I get the following message:

ToggleTerm is lazy loaded, so at the time of your require it isn't available yet.

**Comment by josemiguelo** [comment]
> Why would you want to access the toggleterm like that?
> 
> Btw, you must have `lvim.builtin.terminal.active = true` to have `toggleterm` installed for you. All options related to toggleterm will be presented as `lvim.builtin.terminal.xxx`

Thanks for your response @cpea2506. My use case is this: I make use of the [darkmode](https://github.com/f-person/auto-dark-mode.nvim) plugin along with the [onedarkpro](https://github.com/olimorris/onedarkpro.nvim) theme, so that at a certain time, my MacOS gets dark and so does my theme with 
```console
      auto_dark_mode.setup({
        set_dark_mode = function()
          vim.api.nvim_set_option('background', 'dark')
        end,
        set_light_mode = function()
          vim.api.nvim_set_option('background', 'light')
        end,
      })
```

The thing is that when theme is light, lazygit uses some akward colors, and I figured out it can use an alternative config file for the light theme. 
- for the dark (and default theme): open a terminal with simply `lazygit`
- for the  light theme: open a terminal with `"lazygit" .. string.format(' --use-config-file="%s"', light_theme_path)`

This needs to be computed every time I open the lazygit terminal, not with the builtin `lvim.builtin.terminal` config, because at the very moment the cmd is computed the theme is dark or light and it remains that way the whole nvim session. That's why I make use of `require("toggleterm")` to achieve something like [this](https://github.com/akinsho/toggleterm.nvim#custom-terminal-usage)

In summary: the builtin toggleterm config of lvim is perfect when the cmd is fixed, but doesn't work for dynamic cmds (might be mistaken :))

**Comment by cpea2506** [comment]
> > Why would you want to access the toggleterm like that?
> > Btw, you must have `lvim.builtin.terminal.active = true` to have `toggleterm` installed for you. All options related to toggleterm will be presented as `lvim.builtin.terminal.xxx`
> 
> Thanks for your response @cpea2506. My use case is this: I make use of the [darkmode](https://github.com/f-person/auto-dark-mode.nvim) plugin along with the [onedarkpro](https://github.com/olimorris/onedarkpro.nvim) theme, so that at a certain time, my MacOS gets dark and so does my theme with
> 
> ```
>       auto_dark_mode.setup({
>         set_dark_mode = function()
>           vim.api.nvim_set_option('background', 'dark')
>         end,
>         set_light_mode = function()
>           vim.api.nvim_set_option('background', 'light')
>         end,
>       })
> ```
> 
> The thing is that when theme is light, lazygit uses some akward colors, and I figured out it can use an alternative config file for the light theme.
> 
> * for the dark (and default theme): open a terminal with simply `lazygit`
> * for the  light theme: open a terminal with `"lazygit" .. string.format(' --use-config-file="%s"', light_theme_path)`
> 
> This needs to be computed every time I open the lazygit terminal, not with the builtin `lvim.builtin.terminal` config, because at the very moment the cmd is computed the theme is dark or light and it remains that way the whole nvim session. That's why I make use of `require("toggleterm")` to achieve something like [this](https://github.com/akinsho/toggleterm.nvim#custom-terminal-usage)
> 
> In summary: the builtin toggleterm config of lvim is perfect when the cmd is fixed, but doesn't work for dynamic cmds (I might be mistaken :))

Ah, that makes sense now. I think you can make use of this terminal option.

https://github.com/LunarVim/LunarVim/blob/257e3ff3697a55e7a17eab876fb8d0b85fd203de/lua/lvim/core/terminal.lua#L43-L45

**Comment by josemiguelo** [comment]
@cpea2506 I think I can't. This is how I'm computing the cmd
```console
local function build_lazygit_path()
  local light_theme_path = "~/.config/lazygit/config-light-theme.yml"
  local config_exists = utils.file_exists(light_theme_path)

  if vim.o.background == "light" and config_exists then
    return "lazygit" .. string.format(' --use-config-file="%s"', light_theme_path)
  end

  return "lazygit"
end

lvim.builtin.terminal.execs = {
  {
    build_lazygit_path(),
    "<leader>gg",
    "LazyGit",
    "float"
  },
}
```
and even with this woking, the cmd is only computed AT STARTUP. I need the cmd to be computed EVERY TIME I open lazygit (with `<leader>gg` in this case) so that it can use the config file if needed

**Comment by lvimuser** [comment]
can't you use `on_open`? https://github.com/akinsho/toggleterm.nvim#custom-terminals
Otherwise, you can call `vim.cmd.PackerLoad('toggleterm.nvim')` before the require.

**Comment by josemiguelo** [comment]
Just read the packer docs and understood why it’s been lazy loaded. Maybe we can [prevent](https://github.com/LunarVim/LunarVim/blob/master/lua/lvim/plugins.lua#L240) this so that the plug-in supports this dynamic terminals? Would this be a breaking change? @lvimuser @cpea2506

**Comment by abzcoding** [comment]
@josemiguelo try `on_config_done` 
e.g:
```lua
  local function build_lazygit_path()
    local light_theme_path = "~/.config/lazygit/config-light-theme.yml"
    local config_exists = utils.file_exists(light_theme_path)

    if vim.o.background == "light" and config_exists then
      return "lazygit" .. string.format(' --use-config-file="%s"', light_theme_path)
    end

    return "lazygit"
  end
  lvim.builtin["terminal"].on_config_done = function()
    local terminal = require("toggleterm.terminal").Terminal
    local lazygit = terminal:new { cmd = "lazygit", hidden = true }

    function _lazygit_toggle()
      lazygit:toggle()
    end
    vim.api.nvim_set_keymap("n", "<leader>-", "<cmd>lua _lazygit_toggle()<CR>", { noremap = true, silent = true })
  end
```

for more info check the code for this part
https://github.com/LunarVim/LunarVim/blob/257e3ff3697a55e7a17eab876fb8d0b85fd203de/lua/lvim/core/terminal.lua#L67-L69

**Comment by josemiguelo** [comment]
That worked great. I will start to use this pattern since it seems most of the plug-ins support it. 
thanks for the help @abzcoding @lvimuser @cpea2506
